src/tutorial/function-dispatching/README.md

```diff
@@ -0,0 +1,352 @@
+# Function Dispatching
+
+Function dispatching is something that is fundamental to any Huff contract. Unlike Solidity and Vyper; Huff does not abstract function dispatching. In this section we will go over how dispatching is performed in the other languages, and how you may go about it in Huff.
+
+## What is the problem?
+
+In the evm, contracts are interacted with by sending messages to them. The ABI standard exists as a canonical way to encode these messages, handling how inputs to functions should be encoded. These strict rules are what allow contracts to understand each other. The ABI standard also tells the contract which function the message intends to interact with. This is done by encoding a 4 byte selector at the beginning of the message. This 4 byte selector is the `keccak` of a function's signature. If you have written interfaces in solidity you may not realize that you are just providing your current contract the ability to generate the 4 byte selectors of a foreign contract.
+
+The rest of this section will detail two types of dispatching, linear dispatching and binary search dispatching.
+
+## Linear Dispatching
+
+From reading the above, or from reading a Huff contract before you may have developed some intuition about how the simplest way to perform dispatching is - A linear lookup. This method will extract the function selector from the calldata message, then brute force compare it to every other function in the contract.
+
+The example below shows what a linear dispatcher may look like for a standard ERC20 token:
+
+```huff
+
+// Interface
```

Comment:
these section headers (Interface, Metadata) are a bit opinionated and maybe unecessary for purposes of this example

Reply:
Good call; I'll remove them